server.js

```diff
@@ -1,38 +1,180 @@
+      authorArray.push(newAuthor)
+      await newAuthor.save()
+    })
+
+    booksData.forEach(async (item) => {
+      const newBook = new Book({
+        ...item,
+        authors: authorArray.find((singleAuthor) => singleAuthor.authors === item.authors)
+      })
+      await newBook.save()
+    })
+  }
+  seedDB()
+}
+
+const port = process.env.PORT || 8093
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: 'Service unavailable' })
+  }
+})
+
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+})
+
+// Endpoint to get all books and search for title by query param
+app.get('/books', async (req, res) => {
+  const { title } = req.query
+
+  try {
+    if (title) {
+      const books = await Book.find({ 
+        title: {
+          $regex: new RegExp(title, 'i')
+        }
+      }).populate('authors')
+      res.json(books)
+    } else {
+      const books = await Book.find().populate('authors')
+      res.json(books)
+    }
+  } catch (error) {
+    res.status(404).json({ error: 'Something went wrong', details: 'error' })
+  }
+})
+
+// Endpoint to get bookID by path param
+app.get('/books/:bookId', async (req, res) => {
+  const { bookId } = req.params
+
+  try {
+    const singleBook = await Book.findById(bookId).populate('authors')
+
+    if (singleBook) {
+      res.json(singleBook)
+    } else {
+      res.status(404).json({ error: `Book with id number ${bookId} does not excist`, details: 'error' })
+    }
+  } catch (error) {
+    res.status(400).json({ error: 'Invalid request', details: 'error' })
+  }
+})
+
+// Endpoint to get author by path param for book id
+app.get('/books/:bookId/author', async (req, res) => {
+  const { bookId } = req.params
+
+  try {
+    const singleBook = await (await Book.findById(bookId)).populate('authors')
+    if (singleBook) {
+      const author = await Author.findById(singleBook.authors)
+      res.json(author)
+    } else {
+      res.status(404).json({ error: 'Author not found', details: 'error' })
+    }
+  } catch (error) {
+    res.status(400).json({ error: 'Invalid request', details: 'error' })
+  }
+})
+
+// Endpoint to get author id by path param
+app.get('/authors/:authorID', async (req, res) => {
+  const { authorID } = req.params
+
+  try {
+    const author = await Author.findById(authorID)
+    if (author) {
+      res.json(author)
+    } else {
+      res.status(404).json({ error: `Author with ${authorID} not found`, details: 'error' })
+    }
+  } catch (error) {
+    res.status(400).json({ error: 'Invalid request' })
+  }
+})
+
+// Endpoint to get all books from author id by path param
+app.get('/author/:authorID/books', async (req, res) => {
+  const { authorID } = req.params
+
```

Comment:
It might be nice to also have a query for typing in the author directly, to see if multiple authors of the same name come up. But overall I think you've done great with your paths! 👏